gateway/handle.go

```diff
@@ -753,15 +765,15 @@ func (gw *Handle) extractJobsFromInternalBatchPayload(reqType string, body []byt
 		err := gw.streamMsgValidator(&msg)
 		if err != nil {
 			gw.logger.Errorn("invalid message in request", logger.NewErrorField(err))
-			return nil, errors.New(response.InvalidStreamMessage)
+			return nil, nil, errors.New(response.InvalidStreamMessage)
 		}
+		arctx := gw.authRequestContextForSourceID(msg.Properties.SourceID)
```

Comment:
Can we introduce a simpler `gw.writeKeyForSourceID` method?

Reply:
good idea